cmd/podman/buildfarm/list.go

```diff
@@ -0,0 +1,134 @@
+package buildfarm
+
+import (
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/containers/common/pkg/completion"
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/common/pkg/report"
+	"github.com/containers/podman/v4/cmd/podman/common"
+	"github.com/containers/podman/v4/cmd/podman/registry"
+	"github.com/containers/podman/v4/cmd/podman/validate"
+	"github.com/spf13/cobra"
+)
+
+var (
+	buildfarmLsDescription = `podman buildfarm ls
+
+List all available farms. The output of the farms can be filtered
+and the output format can be changed to JSON or a user specified Go template.`
+	lsCommand = &cobra.Command{
+		Use:               "ls [options]",
+		Aliases:           []string{"list"},
+		Args:              validate.NoArgs,
+		Short:             "List all existing farms",
+		Long:              buildfarmLsDescription,
+		RunE:              list,
+		ValidArgsFunction: completion.AutocompleteNone,
+	}
+
+	// Temporary struct to hold cli values.
+	lsOpts = struct {
+		Filter    []string
+		Format    string
+		Quiet     bool
+		NoHeading bool
+	}{}
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Command: lsCommand,
+		Parent:  buildfarmCmd,
+	})
+	flags := lsCommand.Flags()
+
+	formatFlagName := "format"
+	flags.StringVar(&lsOpts.Format, formatFlagName, "", "Format farm output using Go template")
+	_ = lsCommand.RegisterFlagCompletionFunc(formatFlagName, common.AutocompleteFormat(&farmOut{}))
+
+	flags.BoolVarP(&lsOpts.NoHeading, "noheading", "n", false, "Do not print headers")
+	flags.BoolVarP(&lsOpts.Quiet, "quiet", "q", false, "Print farm output in quiet mode")
```

Comment:
I personally don't like these two options.  The `--format` flag should be used instead.  I know that `machine` has these flags but I think that was a mistake.

Reply:
I was following other list commands that have these flags but removed it here if the preference is to not continue with these in the future.
Fixed